utilities/infra.py

```diff
@@ -286,23 +286,24 @@ def get_pod_container_error_status(pod):
 def get_not_running_pods(pods, filter_pods_by_name=None):
     pods_not_running = []
     for pod in pods:
-        pod_instance = pod.instance
-        if filter_pods_by_name and filter_pods_by_name in pod.name:
-            LOGGER.warning(f"Ignoring pod: {pod.name} for pod state validations.")
-            continue
-        container_status_error = get_pod_container_error_status(pod=pod)
-        if container_status_error:
-            pods_not_running.append({pod.name: container_status_error})
         try:
-            # Waits for all pods in a given namespace to be in final healthy state(running/completed).
-            # We also need to keep track of pods marked for deletion as not running. This would ensure any
-            # pod that was spinned up in place of pod marked for deletion, reaches healthy state before end
-            # of this check
-            if pod_instance.metadata.get("deletionTimestamp") or pod_instance.status.phase not in (
-                pod.Status.RUNNING,
-                pod.Status.SUCCEEDED,
-            ):
-                pods_not_running.append({pod.name: pod.status})
+            pod_instance = pod.instance
+            if filter_pods_by_name and filter_pods_by_name in pod.name:
+                LOGGER.warning(f"Ignoring pod: {pod.name} for pod state validations.")
+                continue
+            container_status_error = get_pod_container_error_status(pod=pod)
+            if container_status_error:
```

Comment:
can be re-written using walrus:
```suggestion

            if container_status_error := get_pod_container_error_status(pod=pod):
```